core/string/ustring.h

```diff
@@ -293,6 +293,8 @@ class String {
 	_FORCE_INLINE_ operator Span<char32_t>() const { return Span(ptr(), length()); }
 	_FORCE_INLINE_ Span<char32_t> span() const { return Span(ptr(), length()); }
 
+	static Span<char32_t> strip_edges_span(const char32_t *sp, uint32_t len, bool left, bool right);
```

Comment:
I have some opinions on where to put these kinds of functions, but I think it's totally fine to park them here for the moment.
However, the first argument should absolutely be a `Span<char32_t>`.

Reply:
I kibnd of hesitated, because you have to construct a span to use it then, like that both string and span have the data to use it directly. But I could see taking a span too.

Reply:
Creating a span is as expensive as creating a pointer and size int - that is, pretty much free. We won't need to worry about that i think :)

Reply:
I think I get it, but eventually I'd like to move a lot of `String` functions to `StringSpan` (or similar) to give callers the chance to avoid `String` entirely if it's not needed. There will probably be many more functions like this, and if they uniformly takes `Span`, that would make it easier in the long run, I think.

Reply:
Changed it to take a span instead.